agent/testing/testlets/http

```diff
@@ -0,0 +1,90 @@
+#!/usr/bin/env bash
+
+# Copyright 2016 Matt Oswalt. Use or modification of this
+# source code is governed by the license provided here:
+# https://github.com/Mierdin/todd/blob/master/LICENSE
+
+check() {
+    echo "Running in check mode"
+    command -v curl >/dev/null 2>&1 || { echo >&2 "curl is not installed. Exiting."; exit 1; }
+    echo "Check mode PASSED"
+    exit 0
+}
+
+if [[ $1 == "check" ]];
+  then
+    check
+fi
+
+# The first argument after the testlet name will ALWAYS be the target. So, $1 is always the target, and all arguments from $2 and up should be passed into the app command
+app_args="$@"
+app_args=${app_args#$0 } # Remove Testlet name
+app_args=${app_args#$1 } # Remove Target
+# at this point, app_args should be all that's left
+
+output=$(curl -s -o /dev/null -w \
+    '\n
+    \nHTTP Code: %{http_code} _ENDA\n
+    \nLookup time: %{time_namelookup} _ENDB\n
+    \nConnect time: %{time_connect} _ENDC\n
+    \nPreXfer time: %{time_pretransfer} _ENDD\n
+    \nStartXfer time: %{time_starttransfer} _ENDE\n
+    \nTotal time: %{time_total} _ENDF\n
+    \nContent type: %{content_type} _ENDG\n
+    \nHTTP Connect: %{http_connect} _ENDH\n
+    \nNumber of Connections: %{num_connects} _ENDI\n
+    \nNumber of Redirections: %{num_redirects} _ENDJ\n
+    \nSize of download: %{size_download} _ENDK\n
+    \nSize of headers: %{size_header} _ENDL\n
+    \nSize of request: %{size_request} _ENDM\n
+    \nSize of upload: %{size_upload} _ENDN\n
+    \nAvg Download Speed: %{speed_download} _ENDO\n
+    \nAvg Upload Speed: %{speed_upload} _ENDP\n
+    \nRedirect time: %{time_redirect} _ENDQ\n
+    \nTime to start transfer: %{time_starttransfer} _ENDR\n
+    \nLast URL Fetched: %{url_effective} _ENDS\n
+    \n' \
+http://$1) #TODO(mierdin): how to handle https?
```

Comment:
FYI This is the one part I'm still working on. Not planning to merge until I fix this